lib/src/test/resources/compileContracts.sh

```diff
@@ -1,47 +1,117 @@
 #!/bin/bash
 
+SRC_V0="src_v0"
+SRC_V1="src_v1"
+SRC_V2="src_v2"
+
+COMPILE_V0="compiled_v0"
+COMPILE_V1="compiled_v1"
+COMPILE_V2="compiled_v2"
+
+COMPILER_BIN_V1="cairo_compilers_v1"
+COMPILER_BIN_V2="cairo_compilers_v2"
+
+get_release_url() {
+  local VERSION=$1
+
+  OS=$(uname)
+  ARCH=$(uname -m)
+
+  local OS_DLOAD="unknown-os"
+  local ARCH_DLOAD="unknown-Aarch"
+  local EXT="tar.gz"
+
+#  if [ "$OS" == "Linux" ]; then
+#    OS_DLOAD="unknown-linux-musl"
+#    ARCH_DLOAD="x86_64"
+#    EXT="tar.gz"
+#  elif [ "$OS" == "Darwin" ]; then
+#    OS_DLOAD="apple-darwin"
+#    ARCH_DLOAD="aarch64"
+#    EXT="tar"
+#  else
+#    echo "Unsupported OS: $OS"
+#    exit 1
+#  fi
+
+    if [ "$ARCH" == "x86_64" ]; then
+      OS_DLOAD="unknown-linux-musl"
+      ARCH_DLOAD="x86_64"
+      EXT="tar.gz"
+    elif [ "$ARCH" == "arm64" ]; then
+      OS_DLOAD="apple-darwin"
+      ARCH_DLOAD="aarch64"
+      EXT="tar"
+      else
+        echo "Unsupported architrecture: $ARCH"
+        exit 1
+    fi
+
+  echo "https://github.com/starkware-libs/cairo/releases/download/v${VERSION}/release-${ARCH_DLOAD}-${OS_DLOAD}.${EXT}"
+}
+
+fetch_compilers() {
+  local VERSION=$1
+  local OUT_DIR=$2
+  local COMPILER_URL=$(get_release_url "$VERSION")
+
+  echo "Fetching compiler v$VERSION binaries..."
+  echo "URL: $COMPILER_URL"
+
+  mkdir -p "$OUT_DIR"
+
+  if [[ $COMPILER_URL == *.tar ]]; then
+    curl -LsSf "$COMPILER_URL" | tar -x -C "$OUT_DIR" || exit 1
+  else
+    curl -LsSf "$COMPILER_URL" | tar -xz -C "$OUT_DIR" || exit 1
+  fi
+
+  COMPILER_PATH=$(pwd)/"$OUT_DIR"/cairo/bin/starknet-compile
+  SIERRA_PATH=$(pwd)/"$OUT_DIR"/cairo/bin/starknet-sierra-compile
+
+  echo "Done!"
+}
```

Comment:
These two should be combined, and simplified. You only need to check what system someone is running, and based on that just download file with requested version.

Reply:
> These two should be combined, and simplified.
As for combining, it can be done although I don't see how removing function makes code better.
> You only need to check what system someone is running, as can be seen by the comment, and based on that just download file with requested version.
I tried that and actually there's a problem with that. Apparently, `macos-latest` on github actions runs macos with x64 arch, so neither of released binaries actually work on CI. It seems there's no full support for arm64 macos as of now: https://github.com/github/roadmap/issues/528 
Not sure what we should do about it, although the issue mentions that it should be in public beta now.

Reply:
> As for combining, it can be done although I don't see how removing function would also simplify the code.

in pseudo code:

```
SYSTEM_NAME=$(uname -s)

if [ "$SYSTEM_NAME" == "Linux" ]; then
  wget https://github.com/starkware-libs/cairo/releases/download/v$VERSION/release-x86_64-unknown-linux-musl.tar.gz
elif [ "$SYSTEM_NAME" == "Darwin" ]; then
  wget https://github.com/starkware-libs/cairo/releases/download/v$VERSION/release-aarch64-apple-darwin.tar
else
  echo "Unsupported platform"
  exit 1
fi
```

this should be enough to have the same functionality as in the marked code.

> I tried that and actually there's a problem with that. Apparently, macos-latest on github actions runs macos with x64 arch, so neither of released binaries actually work on CI. It seems there's no full support for arm64 macos as of now: https://github.com/github/roadmap/issues/528
Not sure what we should do about it, although the issue mentions that it should be in public beta now.

Yup, this is a problem. I see two possible workarounds for now, until gh supports m1s - we can either fetch cairo repo and build it when runners are macos (+ it can be cached), or we can fetch it from somewhere else.